smartpy/losses/classification_losses.py

```diff
@@ -0,0 +1,16 @@
+from smartpy.interfaces.loss import Loss
+
+import theano.tensor as T
+
+
+class NegativeLogLikelihood(Loss):
+    def _loss_function(self, model_output):
+        nll = -T.log(model_output)
+        indices = T.cast(self.target[:, 0], dtype="int32")  # Targets are floats.
+        selected_nll = nll[T.arange(self.target.shape[0]), indices]
```

Comment:
No it isn't. Using `:` will return a matrix instead of a vector.
```python
>>> numpy as np
>>> A = np.diag(range(10))
>>> indices = range(10)
>>> A[:,indices]
array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 2, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 3, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 4, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 5, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 6, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 7, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 8, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 9]])
>>> A[range(10),indices]
array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
```